alembic/script/base.py

```diff
@@ -959,26 +966,40 @@ def _from_path(
 
     @classmethod
     def _list_py_dir(cls, scriptdir: ScriptDirectory, path: str) -> List[str]:
-        if scriptdir.sourceless:
-            # read files in version path, e.g. pyc or pyo files
-            # in the immediate path
-            paths = os.listdir(path)
-
-            names = {fname.split(".")[0] for fname in paths}
-
-            # look for __pycache__
-            if os.path.exists(os.path.join(path, "__pycache__")):
-                # add all files from __pycache__ whose filename is not
-                # already in the names we got from the version directory.
-                # add as relative paths including __pycache__ token
-                paths.extend(
-                    os.path.join("__pycache__", pyc)
-                    for pyc in os.listdir(os.path.join(path, "__pycache__"))
-                    if pyc.split(".")[0] not in names
-                )
-            return paths
-        else:
-            return os.listdir(path)
+        paths = []
+        for root, dirs, files in os.walk(path):
```

Comment:
**mike bayer** (zzzeek) wrote:

it might be a good idea to indicate topdown=True here, even though this is default, to document / assert that the algo here relies upon the topdown walking behavior when it breaks out for non-recursive.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/4457

Reply:
added explicit `topdown=True`